### DISCUSSION TITLE
What type to use for filterQuery/updateQuery

### DISCUSSION THREAD

**Discussion by wasya243** [typegoose/typegoose#850] [Q&A] [answered]
Hello, help me please. Here is sample code:
`const userUpdate: UpdateQuery<IUser> = {     $inc: {},   };`

```
const filter: mongoose.FilterQuery<IOrder> = {
      type: 'order_type',
      _id: orderId,
};
```
Basically it ensures that fields used for queries are the ones present in order or user. What is the code is gonna be for typegoose case?

**Comment by hasezoey** [comment, marked as answer]
> Basically it ensures that fields used for queries are the ones present in order or user. What is the code is gonna be for typegoose case?

the same, replace `IOrder` or `IUser` with the class name instead (for these types nor wrapping in `DocumentType` is needed), at least that is what mongoose is currently set to when you do `YourModel.find({})` and hover over the type (if your editor supports that)

**Comment by wasya243** [comment]
Okey, thank you
